beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -806,8 +819,48 @@ where
                                             request,
                                         };
                                     let delay_key = &entry.get().1;
-                                    self.outbound_substreams_delay
-                                        .reset(delay_key, Duration::from_secs(RESPONSE_TIMEOUT));
+                                    let number_of_received_chunks_for_this_stream: u64 =
+                                        match &entry.get().3 {
+                                            Some(count) => count + 1,
+                                            // There should always be a count here, but if not we can simply set it to 0
+                                            None => 0,
+                                        };
+                                    let protocol = &entry.get().2;
+                                    let maximum_allowed_number_of_chunks_for_this_request: u64 =
```

Comment:
Not sure if there are any drawbacks, but what if instead of storing the number of seen chunks, you store the number of remaining chunks? This way you only request the expected number of chunks once

Reply:
Oh, good idea! Also no need to keep on accessing the request.

Will add this in a new commit. Thanks 🙃

Reply:
@divagant-martian updated. Want to give it another look?

Reply:
@divagant-martian , just pushed the changes. Want to give it another look?

Reply:
this looks good to me, just added a suggestion to make the code a bit cleaner